node-backend/app/controllers/v1/usersController.js

```diff
@@ -1,33 +1,118 @@
+const qs = require(/*eslint-disable  node/no-extraneous-require*/ "qs");
+const yup = require("yup");
+
+const utility = require("../../utils/utility");
 const db = require("../../models/sequelize");
 const Users = db.users;
-
 const jwtToken = require("../../jwtTokenValidation/jwtValidation");
 module.exports.findUsersByOrg = async (req, res) => {
   const authHeader = req.headers["authorization"];
   let userData = await jwtToken.getData(authHeader);
-  Users.findAll({
-    where: { org_id: userData.orgId, role_id: 2 },
-    attributes: [
-      "id",
-      "first_name",
-      "email",
-      "display_name",
-      "profile_image_url",
-      "role_id",
-      "hi5_quota_balance",
-      "org_id",
-    ],
-  })
-    .then((users) => {
-      res.status(200).send({
-        data: users,
-      });
+  const schema = yup.object().shape({
+    org_id: yup.number().typeError({ org_id: "should be a number" }),
+    starts_with: yup.string().typeError({ starts_with: "should be a string" }),
+    limit: yup.number().typeError({ limit: "should be a number" }),
+    offset: yup.number().typeError({ offset: "should be a number" }),
+  });
+  let obj = qs.parse(req.query);
+
+  schema
+    .validate(obj, { abortEarly: false })
+    .then(() => {
+      let limit = 10;
```

Comment:
write a separate function which will return you expected limit and offset. You should make this common function in utils so that other controllers can use this. so for common function you will pass request object. And function will return limit and offset by doing internal computation